app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,44 @@
+class ApplicationsController < ApplicationController
+  def show
+    @application = Application.find(params[:id])
+
+    if params[:search].present? 
+      @pet_search_results = Pet.search_by_name(params[:search])
+      # require 'pry'; binding.pry
+    else
+      @pet_search_results = []
+    end
+  end
+
+  def new
+
+  end
+
+  def create
+    application = Application.new(application_params)
+      if application.save
+        redirect_to "/applications/#{application.id}"
+      else
+        redirect_to "/applications/new"
+        flash[:alert] = "Error: #{error_message(application.errors)}"
```

Comment:
Great job using the inherited `error_message` method from ApplicationController!